src/mongocxx/test/collection_mocked.cpp

```diff
@@ -598,6 +597,10 @@ TEST_CASE("Collection", "[collection]") {
                     }
                 }
 
+                if (expected_hint) {
+                    REQUIRE(options_view["hint"].get_utf8() == expected_hint->get_utf8());
+                }
```

Comment:
Here, and in the other expectations, it might be worth adding:

```
else {
    REQUIRE (!options_view["hint"]);
}
```

To ensure we're not sending an empty hint document when none is given.

Reply:
Done